include/device.h

```diff
@@ -779,10 +779,21 @@ BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
 		(&DEVICE_NAME_GET(dev_name)), level, prio)
 
 #ifdef CONFIG_PM_DEVICE
-#define Z_DEVICE_DEFINE_PM_INIT(dev_name, pm_control_fn)		\
+#define Z_DEVICE_STATE_DEFINE(node_id, dev_name)			\
+	static struct device_state Z_DEVICE_STATE_NAME(dev_name) = {	\
+		.pm = {						\
+			.atomic_flags = ATOMIC_INIT(DT_PROP_OR(	\
+				node_id, wakeup_source, 0) <<		\
+				PM_DEVICE_ATOMIC_FLAGS_WS_CAPABLE_BIT), \
```

Comment:
Automatic enable is not happening now, ENABLE is missing here.

Reply:
that is what is wanted right ? Not enable by default. Will keep it and change the documentation.

Reply:
If we take the path of disabled by default yes, in which case this code is right. Docs need to be updated.